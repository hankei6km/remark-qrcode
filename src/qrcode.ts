import { Plugin, Transformer } from 'unified'
import { Node } from 'unist'
import { Root } from 'mdast'
import { toImageDataURL } from 'mdast-qrcode'

export const remarkQRCode: Plugin = function rehypeImageSalt(): Transformer {
  return async function transformer(tree: Node): Promise<void> {
    await toImageDataURL(tree as Root)
  }
}
