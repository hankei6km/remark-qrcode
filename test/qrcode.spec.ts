import { unified } from 'unified'
import remarkParse from 'remark-parse'
import stringify from 'remark-stringify'
import { remarkQRCode } from '../src/qrcode.js'

describe('remarkQRCode()', () => {
  const f = async (markdown: string): Promise<string> => {
    return await unified()
      .use(remarkParse)
      .use(remarkQRCode)
      .use(stringify, {})
      .freeze()
      .process(markdown)
      .then(
        (file) => {
          return String(file)
        },
        (error) => {
          throw error
        }
      )
  }

  it('should return count of content', async () => {
    expect(await f('#test\n\n![qrcode](qrcode:abcdef123456)')).toMatchSnapshot()
  })
})
