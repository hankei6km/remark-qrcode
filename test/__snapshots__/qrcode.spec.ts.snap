// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remarkQRCode() should return count of content 1`] = `
"\\\\#test

![qrcode](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHQAAAB0CAYAAABUmhYnAAAAAklEQVR4AewaftIAAAKiSURBVO3BQW7sWAwEwSxC979yjpdcPUCQuu3PYUT8wRqjWKMUa5RijVKsUYo1SrFGKdYoxRqlWKMUa5RijVKsUYo1SrFGKdYoFw8l4ZtUTpJwonJHEr5J5YlijVKsUYo1ysXLVN6UhCdUTpLQqZyovCkJbyrWKMUapVijXHxYEu5QuSMJJ0k4UXkiCXeofFKxRinWKMUa5WIYlf+zYo1SrFGKNcrFcEnoVLokdCr/smKNUqxRijXKxYep/CVJ6FSeUPlLijVKsUYp1igXL0vCX6bSJaFTOUnCX1asUYo1SrFGiT/4hyXhROX/pFijFGuUYo1y8VASOpUuCW9S6VROknCicpKEN6l8UrFGKdYoxRrl4pepdEnoVE6ScKLSJaFLQqdyonJHEk6S0Kk8UaxRijVKsUa5+GNUuiS8SaVLwolKl4RO5USlS0Kn8qZijVKsUYo1ysVDKicqdyShU+mS0Kl0SeiScEcSOpU7kvCbijVKsUYp1igXDyXhDpUuCZ1Kl4Q3qXRJOEnCHSq/qVijFGuUYo1y8ZDKEyonKidJeJNKl4RO5Y4kdCpdEjqVJ4o1SrFGKdYo8QcPJOGbVE6ScIdKl4Q7VLokdCpdEjqVNxVrlGKNUqxRLl6m8qYknCShU+mS0KncodIloUvCSRI6lS4JncoTxRqlWKMUa5SLD0vCHSqflIRO5SQJncoTSehU3lSsUYo1SrFGuRgmCZ3KSRI6lZMkPKHyScUapVijFGuUi3WUhE7ljiR8U7FGKdYoxRrl4sNUPkmlS0KXhE7lDpU7kvCbijVKsUYp1igXL0vCNyXhROWbVE6S0Km8qVijFGuUYo0Sf7DGKNYoxRqlWKMUa5RijVKsUYo1SrFGKdYoxRqlWKMUa5RijVKsUYo1yn+xnfrs/7it5AAAAABJRU5ErkJggg==)
"
`;
